src/main/java/by/itech/lab/supplier/service/impl/RetailerServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package by.itech.lab.supplier.service.impl;
+
+import by.itech.lab.supplier.domain.Retailer;
+import by.itech.lab.supplier.dto.CustomerDto;
+import by.itech.lab.supplier.dto.RetailerDto;
+import by.itech.lab.supplier.dto.mapper.RetailerMapper;
+import by.itech.lab.supplier.exception.ResourceNotFoundException;
+import by.itech.lab.supplier.repository.RetailerRepository;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import java.util.Optional;
+
+@Service
+@Transactional
+@Slf4j
+@AllArgsConstructor
+public class RetailerServiceImpl implements by.itech.lab.supplier.service.RetailerService {
+    private final RetailerRepository retailerRepository;
+
+    private final RetailerMapper retailerMapper;
+
+    @Override
+    public Optional<RetailerDto> findById(Long id) {
+        return Optional.of(retailerRepository.findById(id).map(retailerMapper::map)
```

Comment:
Please, extract the result of the dao call to the separate variable and use it here.